src/styles/v2/sing-colors.scss

```
@@ -0,0 +1,674 @@
+/*
+
+## 利用
+
+基本的にはMaterialDesign3のカラーシステムに近い考え方であり
+おおざっぱなロールごと(primary/secondaryなど)のベース彩度+色相は固定し、
+詳細なロール(on-primary/secondary-containerなど)に応じて明度のみ変更する形です
+
+https://m3.material.io/styles/color/roles
+
+## 各ロール
+
+- primary: 
+主要・強調される操作可能部のカラー・ブランドカラーです
+
+- secondary: 
+主要でない操作可能部のカラーです
+
+- tertiary: 
+現時点での利用部分想定としては一時的・未決定部などでプライマリ・セカンダリで表現しきれない場合に利用します
+eg: ノートのプレビュー / 一時的に全ソロのインジケータ / シンガー発音時インジケータ / 失敗がありうる場合の待ちなど
+
+- error:
+エラー・警告です
+
+- neutral
+背景、テキスト、境界線など、UI全体で使用される中立的なカラーです
+ユーザーの注意を引くことなく、背景やサポート的な役割を果たす要素です
+
+- neutral-variant
+背景、セクションの区切り、低コントラストなUI要素などに利用します
+操作可能なUI要素のカラーです
+
+- link
+外部リンクなどのリンクを表すカラーです
+
+## カラースキームの指定
+
+以下のようにCSS変数を指定します
+
+background-color: var(--scheme-color-primary);
+color: var(--scheme-color-on-primary);
+
+## カスタマイズサンプル
+
+OKLab(OKLCH)で指定します
+
+l: 明度 / c: 彩度 / h: 色相
+
+50%明度プライマリを指定:
+  color: oklch(var(--lr-50) var(--primary-c) var(--primary-h));
+
+50%明度プライマリを指定: CSS Variables+相対CSSの場合
+  color: oklch(from var(--primary-key) l calc(var(--lr-50)));
+
+ロールカラー指定: --on-primaryのdarkテーマの値を指定したサンプル
+  --on-primary-dark: oklch(var(--lr-10) var(--primary-c) var(--primary-h));
+
+エイリアス指定: 特定のロール(この場合はneutral)だが明度が違う
+  --sing-piano-key-white-dark: oklch(var(--lr-70) var(--neutral-c) var(--neutral-h));
+
+プライマリと色調を揃えた新規カラー指定: (色相が150度異なる紫になる)
+  color: oklch(from var(--primary-key) l calc(var(--lr-60)) c h calc(var(--primary-c) + 150)));
+
+---
+
+ほとんどの場合、
+
+- role/on-role/role-container/on-role-containerの4つ
+- 必要あればそれぞれのhover/active/focus/disabledの状態
+
+のスタイルを指定することになるため、明度ルールが安定したらmixin化してもいいかも...
+
+eg: 常にホバー時は+5%明度上げるのであれば
+&:hover: oklch(from var(--primary) l calc(...lr-50をlr-55にするmixin));
```

ここだけ急にメモになってるので、メモになってることが分かりやすいようにするのはどうでしょう？
```suggestion
TODO: ほとんどの場合、

- role/on-role/role-container/on-role-containerの4つ
- 必要あればそれぞれのhover/active/focus/disabledの状態

のスタイルを指定することになるため、明度ルールが安定したらmixin化するか検討する。

eg: 常にホバー時は+5%明度上げるのであれば
&:hover: oklch(from var(--primary) l calc(...lr-50をlr-55にするmixin));
```